amass/amass.go

```diff
@@ -247,10 +222,39 @@ loop:
 	return nil
 }
 
-func (e *Enumeration) releaseDomainName(idx int) {
+func (e *Enumeration) periodicChecks(services []core.Service) {
+	done := true
+	for _, srv := range services {
+		if srv.IsActive() {
+			done = false
+			break
+		}
+	}
+	if done {
+		close(e.Done)
```

Comment:
So if any of the services are still active, you don't ever come back and close `e.Done`? Did you intend to have a loop over this?

Reply:
If any of the services are still active, e.Done is not closed, which will prevent the loop from terminating. Once all the services are inactive, then the loop will terminate

Reply:
What loop are you referring to? The range loop will stop once it runs out of elements in the services array.

Reply:
The loop at the end of the Start() method that calls the periodicChecks method

Reply:
Oh, I see now. I should have looked at the file as a whole :smile:.